From 0de3bd01f3a8c570991198afd6cad094f880cfad Mon Sep 17 00:00:00 2001
From: Muthukumar K <muthukumar.kandasamy@intel.com>
Date: Tue, 4 Dec 2012 00:43:32 +0530
Subject: TOMBSTONE in MatroskaExtractor.cpp:673 CHECK_EQ assertion failure

BZ: 65124

Since we donot have the crash reproducible clip, a protective
fix has been added to avoid the crash in failure scenarios.
This will result in graceful exit in error conditions.

Category: AOSP Improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: yes

Orig-Change-Id: Ie25a56b7245aca7a9d1452b678fc70f64716d469

Change-Id: I9fdf434b62aeca88a374aa490276a7f9076a4c8f
Signed-off-by: Muthukumar K <muthukumar.kandasamy@intel.com>
Reviewed-on: http://android.intel.com:8080/82107
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 .../libstagefright/matroska/MatroskaExtractor.cpp  |   10 +++++++---
 1 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/media/libstagefright/matroska/MatroskaExtractor.cpp b/media/libstagefright/matroska/MatroskaExtractor.cpp
index dc38108..472a4e6 100755
--- a/media/libstagefright/matroska/MatroskaExtractor.cpp
+++ b/media/libstagefright/matroska/MatroskaExtractor.cpp
@@ -681,13 +681,17 @@ MatroskaExtractor::MatroskaExtractor(const sp<DataSource> &source)
     }
 
     ret = mSegment->ParseHeaders();
-    CHECK_EQ(ret, 0);
+    if (ret < 0) {
+        LOGE("Parse headers error %d", ret);
+        delete mSegment;
+        mSegment = NULL;
+        return;
+    }
 
     long len;
     ret = mSegment->LoadCluster(pos, len);
-    CHECK_EQ(ret, 0);
-
     if (ret < 0) {
+        LOGE("Cluster loading =error %d", ret);
         delete mSegment;
         mSegment = NULL;
         return;
-- 
1.7.4.1

